app/src/main/java/com/ankihighlights/android/viewmodel/HighlightViewModel.kt

```diff
@@ -0,0 +1,57 @@
+package com.ankihighlights.android.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.ankihighlights.android.model.HighlightData
+import com.ankihighlights.android.model.HighlightResponse
+import com.ankihighlights.android.network.RetrofitHighlightClient
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
+
+@HiltViewModel
+class HighlightViewModel @Inject constructor() : ViewModel() {
+
+    fun startListeningForHighlights() {
+        viewModelScope.launch {
+            while (true) {
+                // Placeholder for highlight event listening logic
+                delay(5000) // Wait for 5 seconds before checking again
+            }
+        }
+    }
+
+    fun sendHighlightToApi(word: String, context: String, timestamp: Long) {
```

Comment:
This should be in a service, not a ViewModel. ViewModels are for dealing with logic related to the view. 

The sudo-hierarchy here is DataSource (Retrofit) -> Repository -> Service (Optional) -> ViewModel -> View.
So basically move all the code back one file-naming-convention, and in the ViewModel it's hopefully something like `highlightService.sendHighlightToApi()`.

Follow this NowInAndroid link to see the relationship between View -> ViewModel -> Repository, using `feedUiState` as an example: https://github.com/android/nowinandroid/blob/420076f01aadd61f650b6848cd23643ca59226b8/feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/BookmarksViewModel.kt#L47

Reply:
This one gave me like 3+ hours of progress and regress unraveling and re-raveling my understanding of these patterns lolol. But trying to make sense of Nia is where it led too. 

so while https://github.com/meisei4/ankihighlights-android/pull/1/commits/6d0cb7bdf0c1cd32964cf4c6a66706820d6bc582 Does in a way solve the problem (by introducing a Repository class), it also introduced a ton more problems because I was also trying to learn Hilt and the Fake Repos and how Nia uses Dependency ALONE to achieve mocking (which has a much nicer feeling than trying to use a TEST-only mocking library)